1.7.0 Changelog (#14582)
Chia-Network/chia-blockchain: CHANGELOG.md

- Remove redundant check for duplicate outputs by @AmineKhaldi in https://github.com/Chia-Network/chia-blockchain/pull/14195
- check_removals() doesn't need to be async by @AmineKhaldi in https://github.com/Chia-Network/chia-blockchain/pull/14199
- Remove redundant check for maximum block CLVM cost by @AmineKhaldi in https://github.com/Chia-Network/chia-blockchain/pull/14212
- Remove redundant check for duplicate inputs by @AmineKhaldi in https://github.com/Chia-Network/chia-blockchain/pull/14272
- bump `chia_rs` dependency to `0.2.0` @arvidn in https://github.com/Chia-Network/chia-blockchain/pull/14289
- Update version of `clvm_tools_rs` to `0.1.30` by @prozacchiwawa in https://github.com/Chia-Network/chia-blockchain/pull/14327
- Use better check that we are on mainnet when deciding to use default Chia DNS server by @aqk in https://github.com/Chia-Network/chia-blockchain/pull/14254
- Remove conflicting TXs before adding SpendBundle to Mempool in add_spend_bundle by @aqk in https://github.com/Chia-Network/chia-blockchain/pull/14292
- Try each Chia DNS Server in list before trying introducers by @aqk in https://github.com/Chia-Network/chia-blockchain/pull/14253

> - optimize mempool's potential cache by @arvidn in https://github.com/Chia-Network/chia-blockchain/pull/14297

Reviewer: I could go either way. I agree that it's pretty obscure to the average user

> - optimize mempool's potential cache by @arvidn in https://github.com/Chia-Network/chia-blockchain/pull/14297

Reviewer 2: I suggest we keep this

- server: Fix invalid attribute accesses in `WSChiaConnection` by @xdustinface in https://github.com/Chia-Network/chia-blockchain/pull/14051
- fix coin selection cli decimal handling & splitting issue by @jack60612 in https://github.com/Chia-Network/chia-blockchain/pull/13908
- Make `--number-of-coins` required for the `chia coins split` CLI by @altendky in https://github.com/Chia-Network/chia-blockchain/pull/14072
- fix warning log in header validation by @arvidn in https://github.com/Chia-Network/chia-blockchain/pull/13944
- revert delayed pool config handling (#12795, #13154) by @altendky in https://github.com/Chia-Network/chia-blockchain/pull/14090
- handle incoming unknown capabilities by @altendky in https://github.com/Chia-Network/chia-blockchain/pull/14093
- replacing transactions in the mempool is normal behavior, not a warning by @arvidn in https://github.com/Chia-Network/chia-blockchain/pull/14089
- don't throw unnecessary exception on peer connect by @almogdepaz in https://github.com/Chia-Network/chia-blockchain/pull/14108
- Return existing CAT wallet instead of raising by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14101

> - server: Resolve peers in harvester and timelord startup also by @xdustinface in https://github.com/Chia-Network/chia-blockchain/pull/14161

Reviewer 2: This fixed a bug - I'll note the bug in the copy

- Fix fee and unit amounts when coin splitting or viewing coins by @jack60612 in https://github.com/Chia-Network/chia-blockchain/pull/14172
- disallow empty SpendBundles in the mempool by @arvidn in https://github.com/Chia-Network/chia-blockchain/pull/14305
- Check for valid timeout and log error  by @emlowe in https://github.com/Chia-Network/chia-blockchain/pull/14304
- add rate limits numbers to none response type by @almogdepaz in https://github.com/Chia-Network/chia-blockchain/pull/14184
- Fixed condition to check for wallet_id by @steppsr in https://github.com/Chia-Network/chia-blockchain/pull/14306
- fix offer cancel balance by @trepca in https://github.com/Chia-Network/chia-blockchain/pull/14342
- Fix bulk tx by @ytx1991 in https://github.com/Chia-Network/chia-blockchain/pull/14372
- Make the `--id` flag on cancel_offer required by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14376
- Fix target puz hash index for bulk mint nfts with airdrop by @geoffwalmsley in https://github.com/Chia-Network/chia-blockchain/pull/14370
- Fix chia wallet coins cli bugs by @jack60612 in https://github.com/Chia-Network/chia-blockchain/pull/14388

> - Fixed wallet state if db is rolled back by @wjblanke in https://github.com/Chia-Network/chia-blockchain/pull/14384

Reviewer 3: ```suggestion
- Fixed wallet DB issues resulting when there are unexpected failures during syncing @wjblanke in https://github.com/Chia-Network/chia-blockchain/pull/14384
```

> - Add some extra checks for size and existence to notification backend by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14398

Reviewer 3: Notifications are new, no need to list out bug fixes for them.

```suggestion
```

> - fix sending of the `new_on_chain_notification` state change message by @paninaro in https://github.com/Chia-Network/chia-blockchain/pull/14410

Reviewer 3: Notifications are new, no need to list out bug fixes for them.

```suggestion
```

> - Add a persitent id table to notification store by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14418

Reviewer 3: Notifications are new, no need to list out bug fixes for them.

```suggestion
```

> - Check incoming notification hints by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14450

Reviewer 3: Notifications are new, no need to list out bug fixes for them.

```suggestion
```

> - Quex.offer mod tests by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14478

Reviewer 3: This is either internal or needs explanation.

```suggestion
```

Reviewer 3: I guess this would be @Quexington as well.

Reviewer 2: Removed this line and added a new one about offer updates in general

> - Purposefully break offer compression for older versions by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14535

Reviewer 3: I think some other words might make this useful to users.  @Quexington?

and this project does not yet adhere to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
for setuptools_scm/PEP 440 reasons.

## 1.7.0 Chia blockchain 2023-02-15

### Added
- New `chia wallet coins` CLI and RPCs for listing, splitting, and combining coins
- New on-chain notification for offers, specifically designed for NFT offers
- New full node dynamic fee estimator (`chia show -f` and `get_fee_estimate`)
- Add infrastructure for soft fork at block 3630000

> - Add gzip support to Datalayer download client by @Chida82 in https://github.com/Chia-Network/chia-blockchain/pull/14320
> - Add proxy support to Datalayer download client by @Chida82 in https://github.com/Chia-Network/chia-blockchain/pull/14281

Reviewer 3: ```suggestion
- Add gzip support to DataLayer download client by @Chida82 in https://github.com/Chia-Network/chia-blockchain/pull/14320
- Add proxy support to DataLayer download client by @Chida82 in https://github.com/Chia-Network/chia-blockchain/pull/14281
```

- Add `get_timestamp_for_height` Wallet RPC for converting heights to timestamps by @Quexington in https://github.com/Chia-Network/chia-blockchain/pull/14357
- Add `tools/legacy_keyring.py` to allow migration from the removed old key storage format.  Available only from source installations. by @xdustinface in https://github.com/Chia-Network/chia-blockchain/pull/13947
- Add Arch Linux to install-gui.sh script by @DaOneLuna in https://github.com/Chia-Network/chia-blockchain/pull/14059
- Add a `daemon_heartbeat` setting to config.yaml by @emlowe in https://github.com/Chia-Network/chia-blockchain/pull/13886
- add `trusted_max_subscribe_items` and `wallet:trusted_peers` to config.yaml by @almogdepaz in https://github.com/Chia-Network/chia-blockchain/pull/14188

> - Nft bulk transfer by @ytx1991 in https://github.com/Chia-Network/chia-blockchain/pull/14329

Reviewer 3: ```suggestion
- NFT bulk transfer and DID assignment wallet RPCs by @ytx1991 in https://github.com/Chia-Network/chia-blockchain/pull/14329
```
